 **Step-1 :**

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

**Step -2 : Read the data**

In [3]:
file_path="C:\\Users\\Greeshma\\OneDrive\\Naresh IT\\files\\Salary_Data - Salary_Data (1).csv"
df=pd.read_csv(file_path)
df

YearsExperience  Salary
0               1.1   39343
1               1.3   46205
2               1.5   37731
3               2.0   43525
4               2.2   39891
5               2.9   56642
6               3.0   60150
7               3.2   54445
8               3.2   64445
9               3.7   57189
10              3.9   63218
11              4.0   55794
12              4.0   56957
13              4.1   57081
14              4.5   61111
15              4.9   67938
16              5.1   66029
17              5.3   83088
18              5.9   81363
19              6.0   93940
20              6.8   91738
21              7.1   98273
22              7.9  101302
23              8.2  113812
24              8.7  109431
25              9.0  105582
26              9.5  116969
27              9.6  112635
28             10.3  122391
29             10.5  121872

In [4]:
df.size

60

In [5]:
df.shape

(30, 2)

In [7]:
df.isnull()

YearsExperience  Salary
0             False   False
1             False   False
2             False   False
3             False   False
4             False   False
5             False   False
6             False   False
7             False   False
8             False   False
9             False   False
10            False   False
11            False   False
12            False   False
13            False   False
14            False   False
15            False   False
16            False   False
17            False   False
18            False   False
19            False   False
20            False   False
21            False   False
22            False   False
23            False   False
24            False   False
25            False   False
26            False   False
27            False   False
28            False   False
29            False   False

In [8]:
df.isnull().sum()

YearsExperience    0
Salary             0
dtype: int64

In [9]:
df.columns

Index(['YearsExperience', 'Salary'], dtype='object')

**Step-3 : Divide the data into two parts input(X) and output(y)**

In [13]:
X=df.drop('Salary',axis=1)
X

YearsExperience
0               1.1
1               1.3
2               1.5
3               2.0
4               2.2
5               2.9
6               3.0
7               3.2
8               3.2
9               3.7
10              3.9
11              4.0
12              4.0
13              4.1
14              4.5
15              4.9
16              5.1
17              5.3
18              5.9
19              6.0
20              6.8
21              7.1
22              7.9
23              8.2
24              8.7
25              9.0
26              9.5
27              9.6
28             10.3
29             10.5

In [14]:
y=df['Salary']
y

0      39343
1      46205
2      37731
3      43525
4      39891
5      56642
6      60150
7      54445
8      64445
9      57189
10     63218
11     55794
12     56957
13     57081
14     61111
15     67938
16     66029
17     83088
18     81363
19     93940
20     91738
21     98273
22    101302
23    113812
24    109431
25    105582
26    116969
27    112635
28    122391
29    121872
Name: Salary, dtype: int64

**Step-4 :**
    
   - divide X into X_train and X_test
    
   - divide y into y_train and y_test
    

In [21]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,
                                                   y,
                                                   random_state=1234,
                                                   test_size=0.30)


In [22]:
X_train.shape,X_test.shape

((21, 1), (9, 1))

In [23]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,
                                                   y,
                                                   random_state=42,
                                                   test_size=0.30)


In [24]:
X_train.shape,X_test.shape

((21, 1), (9, 1))

In [26]:
X_train

YearsExperience
0               1.1
4               2.2
16              5.1
5               2.9
13              4.1
11              4.0
22              7.9
1               1.3
2               1.5
25              9.0
3               2.0
21              7.1
26              9.5
18              5.9
29             10.5
20              6.8
7               3.2
10              3.9
14              4.5
19              6.0
6               3.0

In [27]:
X_test

YearsExperience
27              9.6
15              4.9
23              8.2
17              5.3
8               3.2
9               3.7
28             10.3
24              8.7
12              4.0

In [28]:
y_train

0      39343
4      39891
16     66029
5      56642
13     57081
11     55794
22    101302
1      46205
2      37731
25    105582
3      43525
21     98273
26    116969
18     81363
29    121872
20     91738
7      54445
10     63218
14     61111
19     93940
6      60150
Name: Salary, dtype: int64

In [29]:
y_test

27    112635
15     67938
23    113812
17     83088
8      64445
9      57189
28    122391
24    109431
12     56957
Name: Salary, dtype: int64

In [25]:
y_train.shape,y_test.shape

((21,), (9,))

**Step-5 :**

**Model development**


In [30]:
from sklearn.linear_model import LinearRegression
LR=LinearRegression()
LR.fit(X_train,y_train)

LinearRegression()

**Model Predictions**

In [33]:
y_predictions=LR.predict(X_test)
y_predictions


array([115573.62288352,  71679.93878159, 102498.90847018,  75415.57147111,
        55803.4998511 ,  60473.04071301, 122110.98009019, 107168.44933209,
        63274.76523015])

**step-6**
**MOdel evaluation**

In [34]:
from sklearn.metrics import r2_score,mean_squared_error

In [39]:
R2=r2_score(y_test,y_predictions)
MSE=mean_squared_error(y_test,y_predictions)
RMSE=np.sqrt(MSE)
print("R-square:",R2)
print("MSE:",MSE)
print("RMSE:",RMSE)

R-square: 0.9414466227178214
MSE: 37784662.46621314
RMSE: 6146.92300799458


**Step-7:Finding coefficients and intercept**

In [38]:
df.columns

Index(['YearsExperience', 'Salary'], dtype='object')

In [36]:
LR.coef_

array([9339.08172382])

In [37]:
LR.intercept_

25918.438334893202

**The linear regression equation is :** 
   - y=intercept + coefficient * Columns
   - y=25918.438334893202+9339.08172382*'YearsExperience'